MPRGWrapper.h

```diff
@@ -0,0 +1,1073 @@
+      //  algorithms and 0.0 for deterministic algorithms.
+      //
+      //  If a non-NULL warnings argument is received, it will be populated
+      //  as applicable.
+      //<
+
+   virtual EcefLocus imageToProximateImagingLocus(
+      const ImageCoord& imagePt,
+      const EcefCoord& groundPt,
+      double desiredPrecision = 0.001,
+      double* achievedPrecision = NULL,
+      WarningList* warnings = NULL) const;
+      //> This method, for the given imagePt (line, sample in full image space
+      //  pixels), returns the position and direction of the imaging locus
+      //  nearest the given groundPt (x,y,z in ECEF meters).
+      //
+      //  Note that there are two opposite directions possible.  Both are
+      //  valid, so either can be returned; the calling application can convert
+      //  to the other as necessary.
+      //
+      //  Iterative algorithms will use desiredPrecision, in meters, as the
+      //  convergence criterion for the locus position, otherwise it will be
+      //  ignored.
+      //
+      //  If a non-NULL achievedPrecision argument is received, it will be
+      //  populated with the actual precision, in meters, achieved by iterative
+      //  algorithms and 0.0 for deterministic algorithms.
+      //
+      //  If a non-NULL warnings argument is received, it will be populated
+      //  as applicable.
+      //<
+
+   virtual EcefLocus imageToRemoteImagingLocus(
+      const ImageCoord& imagePt,
+      double desiredPrecision = 0.001,
+      double* achievedPrecision = NULL,
+      WarningList* warnings = NULL) const;
+      //> This method, for the given imagePt (line, sample in full image space
+      //  pixels), returns the position and direction of the imaging locus
+      //  at the sensor.
+      //
+      //  Note that there are two opposite directions possible.  Both are
+      //  valid, so either can be returned; the calling application can convert
+      //  to the other as necessary.
+      //
+      //  Iterative algorithms will use desiredPrecision, in meters, as the
+      //  convergence criterion for the locus position, otherwise it will be
+      //  ignored.
+      //
+      //  If a non-NULL achievedPrecision argument is received, it will be
+      //  populated with the actual precision, in meters, achieved by iterative
+      //  algorithms and 0.0 for deterministic algorithms.
+      //
+      //  If a non-NULL warnings argument is received, it will be populated
+      //  as applicable.
+      //
+      //  Notes:
+      //
+      //  The remote imaging locus is only well-defined for optical sensors.
+      //  It is undefined for SAR sensors and might not be available for
+      //  polynomial and other non-physical models.  The
+      //  imageToProximateImagingLocus method should be used instead where
+      //  possible.
+      //<````````````````````````````````````````````````````````1111111111111111111111111111111111111111111111111111111111111111111111111111
```

Comment:
Remove the spurious text in this comment line

Reply:
done